pkg/api/server.go

```diff
@@ -328,27 +324,35 @@ func GetVersion() (string, string, string) {
 	return version, githash, buildstamp
 }
 
-func makeTLSConfig() *tls.Config {
-	cert, err := ioutil.ReadFile(paths.GetSSLCert())
+func makeTLSConfig(c *config.Instance) (*tls.Config, error) {
+	c.InitTLS()
+	certFile, keyFile := c.GetTLSFiles()
+
+	if certFile == "" || keyFile == "" {
```

Comment:
If a user supplies only one of the cert/key do we count that as a `broken` TLS configuration?
Should stash in that case fallback to http as it does now (dev and pr) or panic with an error instead?
```golang
	if certFile == "" && keyFile == "" {
```
for example would fallback to http if **none** of the key/cert file are provided and panic if only one of those is provided or either is a non valid certificate file.

The PR tests and looks ok.

Reply:
Changed so that if either SSL file is present, then both must be present, otherwise an error is returned.